### ISSUE TITLE
OAuth2: Wrong "Max-Age" attribute for cookies

### ISSUE THREAD

**Issue by mkauf** [envoyproxy/envoy#28726]
The OAuth2 filter wants to create cookies that expire at the same time as the access token. But it uses a wrong value for the "Max-Age" attribute of the cookies.

"Max-Age" is a relative time (number of seconds until the cookie expires), but the OAuth2 filter uses an absolute timestamp (seconds since the epoch).

*Example:*
```
Set-Cookie: OauthHMAC=ZWU2ZDIwOWU0ODQ1Mjk5ZDczODA3MWI3ODQxN2I3YTg0MDg5YmE1MzA4ZTk5MjBhNmY0M2E3ODBlMjZiNDMzMw==;version=1;path=/;Max-Age=1690815337;secure;HttpOnly
Set-Cookie: OauthExpires=1690815337;version=1;path=/;Max-Age=1690815337;secure;HttpOnly
```

"Max-Age=1690815337" means that the cookie will expire in the year 2077...

*Expected bugfix:*
Use correct values in the "Max-Age" attribute, or use the "Expires" attribute instead if you want to use an absolute timestamp (this needs an HTTP timestamp string, not just a number of seconds).

**Comment by htuch** [comment]
@derekargueta @mattklein123

**Comment by mkauf** [comment]
This has already been fixed by #26764
